### ISSUE TITLE
Issue with esphome gateway and passive ble_monitor with Govee H5075

### ISSUE THREAD

**Issue by jchang10** [custom-components/ble_monitor#641]
https://github.com/myhomeiot/esphome-components/issues/5

Hi Ernst. @myhomeiot recommended I should post here about this above issue. I am a newbie to BLE, and I just have this one Govee H5075 sensor I was experimenting with. Basically, the Govee worked great with just the ble_monitor in HA. I thought I would see if I could make the ESP gateway work, too. However, there is a difference between the way the two sees a data packet. Here is an example.

ble_monitor_packet = b'\x04>+\x02\x01\x00\x00\x11l$8\xc1\xa4\x1f\r\tGVH5075_6C11\x03\x03\x88\xec\x02\x01\x05\t\xff\x88\xec\x00\x032\x9cd\x00\xd9'

gateway_packet = b'\x04>F\x02\x01\x00\x00\x11l$8\xc1\xa4:\r\tGVH5075_6C11\x03\x03\x88\xec\x02\x01\x05\t\xff\x88\xec\x00\x01-nd\x00\x1a\xffL\x00\x02\x15INTELLI_ROCKS_HWPu\xf2\xff\xc2\xb3'

The gateway packet is a bit longer and has some unrecognized stuff at the end. The problem is that the passive code reads the beginning parts fine, but then gets to the unrecognized bit and then drops the entire packet on the floor. Thus, the Govee is never able to send any data at all with the gateway.

Comparing the two data packets above, it looks like the gateway is picking up some extra data on the end. @myhomeiot says that BLE works with HCI packets and he is working with a BLE packet I guess. Thus, he is, in effect, wrapping the BLE packet as an HCI packet and sending it to HA as is. Not sure where the error might be, then.

https://github.com/custom-components/ble_monitor/blob/43341e01c850f79e690d1b13a1e204835bbc44d2/custom_components/ble_monitor/ble_parser/__init__.py#L105-L107

This is the line where passive overwrites the good sensor values with the unrecognized values, and things go bad from there. bad_struct over writes the good man_spec_data on the above line.
bad_struct = b'\x1a\xffL\x00\x02\x15INTELLI_ROCKS_HWPu\xf2\xff\xc2'

I dont understand things well enough to know where and if a fix could occur for this one device. I figure if this one device has this issue, potentially others will too when working with the ESP gateway. I spent a day sniffing this out, so I figure I would not let it go to waste and see if others might make sense of all this and have suggestions.

Thanks!
Jae

**Comment by Ernst79** [comment]
If I convert your packets to a hci string, you have the following strings. 

```
043e2b02010000116c2438c1a41f0d09475648353037355f36433131030388ec02010509ff88ec0003329c6400d9
043e4602010000116c2438c1a43a0d09475648353037355f36433131030388ec02010509ff88ec00012d6e64001aff4c000215494e54454c4c495f524f434b535f48575075f2ffc2b3
```

The first converts fine, where this "manufacturer specific data" part `09ff88ec0003329c6400` is used to get the measurements
```
result  ({'firmware': 'Govee', 'temperature': 20.9564, 'humidity': 56.4, 'battery': 100, 'rssi': -39, 'mac': 'A4C138246C11', 'type': 'H5072/H5075', 'packet': 'no packet id', 'data': True}, None)
```

The second results in an error and contains two "manufacturer specific data" parts. 

```
09ff88ec00012d6e6400
1aff4c000215494e54454c4c495f524f434b535f48575075f2ffc2
```

What goes wrong in the parser is that it the second part is overwriting the first part. 

I have to do some reading, if this is allowed (and hence should be fixed in BLE monitor) or if this is not allowed (and hence should be fixed in ESP gateway).

**Comment by Ernst79** [comment]
Should be fixed in 7.3.1-beta (enable beta releases in HACS to install it)

**Comment by jchang10** [comment]
Wow. @Ernst79 Thanks. That was quick!

A quick question, I had sprinkled some print lines in the BleParser.parse_data() function. I noticed that most of the time it gets called, it simply returns None,None for sensor_data and tracker_data. It looks like a very fast busy loop. In fact, print lines hardly do any good, because of the number of times this function gets called, seemingly infinitely, to do nothing at all - ie return None,None.

I just checked this beta version 7.3.1-beta, it does seem better behaved. There is now a spurt of 100 calls, a pause, 100 calls, a pause, about every 10 seconds or so now. Before it was just continuously running.

Just out of curiosity, can you explain what this is about? Is it just scanning for device activity? The calls to parse_data() do not seem to correlate with 'hcidump' for example. I just cant imagine that much bluetooth activity locally to justify that number of calls.

**Comment by Ernst79** [comment]
I’m not sure why you notice a difference. Ble monitor is continuously scanning. Most messages will return None, None, as you have way more Bluetooth devices that are not sending useful data, like phones, etc. Also supported sensors often send additional data that can’t be used. Depending on the device this can vary from a message once per second till once per hour.

**Comment by Ernst79** [comment]
Released as final in 7.3.8
